samples/ec_generate.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+
+#   Copyright (C) 2019 Atte Pellikka <atte.pellikka@gmail.com>
+#
+# This file is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA.
+
+from PyKCS11 import *
+from asn1crypto.keys import ECDomainParameters, NamedCurve
+
+pkcs11 = PyKCS11.PyKCS11Lib()
+pkcs11.load()
+
+slot = pkcs11.getSlotList(tokenPresent=True)[0]
+
+session = pkcs11.openSession(slot, CKF_SERIAL_SESSION | CKF_RW_SESSION)
+session.login("1234")
+
+key_id = (0x22,)
+label = 'test'
+
+# Select the curve to be used for the keys
+curve = 'secp256r1'
+
+# Setup the domain parameters, unicode conversion needed for the curve string
+domain_params = ECDomainParameters(name='named', value=NamedCurve.unmap(unicode(curve)))
```

Comment:
Why do you use NamedCurve.unmap() and not NamedCurve() only?

The following code works for Python 2 and Python 3:
curve = u'secp256r1'
value=NamedCurve(curve)